### DISCUSSION TITLE
Support postgres schema in the configuration

### DISCUSSION THREAD

**Discussion by winston0410** [ente-io/ente#2603] [Enhancements]
I would like to use specific schema instead of `public` to connect to my Postgres database. I guess we just need to accept the schema value here:
https://github.com/ente-io/ente/blob/99c6c22fe16c16e0ee75c2f4beebdbaae1d53583/server/pkg/utils/config/config.go#L111

And I notice the connection detail is immidiately used here:
https://github.com/ente-io/ente/blob/99c6c22fe16c16e0ee75c2f4beebdbaae1d53583/server/cmd/museum/main.go#L768

If you are happy with a key say, `db.schema` in the config, I can help with that with a PR

**Comment by mnvr** [comment]
I'm not fully aware of the details, but from my understanding there is no "schema" parameter for the connection string, there is instead a possibility of specifying a search path. So calling it schema, while helpful, would be inaccurate.

Perhaps a better option to cover the long tail of such configurability is to add a `db.extra` option, a string that is appended verbatim to the connection string inside `GetPGInfo`. That ways, (a) there is no behaviour change for existing users, and (b) there is no if statement in the code (we try to avoid branches whenever possible in the server side code).

Only impact for existing setups would be a trailing space in the connection string, but I think that should be fine (but please do verify it).

**Reply by winston0410** [reply]
Yup, I have been used `search_path` in the past to define the schema, and I think `db.extra` to handle all string parameter makes sense. I think both trailing and leading space would not cause any issue, if we want we can also remove them with `string.trimSpace` I guess
